### DISCUSSION TITLE
如何获取 compositionStartTime

### DISCUSSION THREAD

**Discussion by rpsh** [Tencent/libpag#2609] [Q&A] [answered]
在获取 replaceImage 所在的 PAGImageLayer 的信息时候， `startTime` 获取的值跟实际播放时出现的时间不对齐，发现还有一个 `compositionStartTime` ，注意到 `compositionStartTime + startTime` 后是实际 replaceImage 播放的位置。
请问如何获取这个 `compositionStartTime` 
（截图来自本地 PAGViewer 播放器的信息面板）
<img width="511" alt="image" src="https://github.com/user-attachments/assets/3ab23033-b5c0-4f91-9bae-5b6e7c9f43ec">

**Comment by kevingpqi123** [comment, marked as answer]
PAGLayer 上面有一个 startTime 的接口， 获取到的是当前图层相对于父级的起始时间，PAGFile 是一个树状渲染结果，如果需要获取相对于根节点的起始时间， 再调用 localTimeToGlobal， 建议多看下 API 文档， API 接口注释写的很详细
